Video Playback Fixes

mo123 opened the pull request:
Fix some realmedia video files can't play.
~~Fix video playback from GoPro cameras.~~
Disable caching for mp4 files, they might have bad a/v interleaving and will cause cache thrashing.

<!--- Provide a general summary of your change in the Title above -->

## Description
<!--- Describe your change in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

MartijnKaijser commented:
jenkins build this please

a1rwulf commented on xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp:
```diff
@@ -1470,6 +1470,13 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
+        {
+          stream->SetDiscard(AVDISCARD_ALL);
+          pStream->discard = AVDISCARD_ALL;
+        }        
```
There's already a similar pull-request open:
https://github.com/xbmc/xbmc/pull/12579

FernetMenta replied:
check my comments on this PR

Rechi replied:
Superceded by #13095

FernetMenta commented on xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamFile.cpp:
```diff
@@ -82,12 +82,17 @@ bool CDVDInputStreamFile::Open()
-      content == "video/x-matroska-3d")
+  if (content == "video/mp4")
+  {
+    flags |= READ_NO_CACHE;
```
I don't think we want to disable caching in general for mp4

mo123 committed:
Disable caching for mp4 files

Disable caching for mp4 files, they might have bad a/v interleaving and will cause cache thrashing.

mo123 committed:
Video: Fix some realmedia video files can't play 

Fix some realmedia video files can't play.

Rechi commented:
@mo123 please share examples of realmedia videos and mp4s which aren't working correctly.

mo123 commented:
Hi

Most of the changes have been added by other PR's now.
I don't have RM test videos to share and the PR is probably not needed anymore.
I will open a new PR if there are still problems with RM videos in the future.
